mirrord/operator/src/crd.rs

```diff
@@ -51,22 +58,20 @@ impl TargetCrd {
         target_config
             .path
             .as_ref()
-            .map_or_else(|| TARGETLESS_TARGET_NAME.to_string(), Self::target_name)
-    }
-
-    pub fn name(&self) -> String {
-        self.spec
-            .target
-            .as_ref()
-            .map(Self::target_name)
-            .unwrap_or(TARGETLESS_TARGET_NAME.to_string())
+            .map_or_else(|| TARGETLESS_TARGET_NAME.to_string(), Self::urlfied_name)
     }
 }
 
 impl From<TargetCrd> for TargetConfig {
     fn from(crd: TargetCrd) -> Self {
         TargetConfig {
-            path: crd.spec.target,
+            path: Some(
+                crd.spec
+                    .target
+                    .known()
+                    .expect("[BUG] Unknown target should never be seen here!")
```

Comment:
Should be an error, so `TryFrom` instead of `From`

Reply:
Changed. Implemented a couple of `TryFrom`s.